Firebase/Core/FIRApp.m

```diff
@@ -393,6 +408,12 @@ + (void)sendNotificationsToSDKs:(FIRApp *)app {
   [[NSNotificationCenter defaultCenter] postNotificationName:kFIRAppReadyToConfigureSDKNotification
                                                       object:self
                                                     userInfo:appInfoDict];
+
+  // This is the new way of sending information to SDKs.
+  // TODO: Do we want this on a background thread, maybe?
```

Comment:
Yes.  Is there any downside to background threads?

Should each library have its own thread, so a long running library configuration doesn't block an urgent library configuration?

Reply:
The reason I didn't add it to a background thread immediately is because I wasn't sure if this would affect current functionality or cause any bugs. The notification is fired on the main thread and I'm fairly certain there are SDKs that do work without offloading to a background thread.

I agree we should move it to a background thread, but I would prefer to move over SDKs first and ensure they are okay to be configured on a background thread before blindly doing so.